### ISSUE TITLE
[Question] Skipping checksum verification on install or update

### ISSUE THREAD

**Issue by OwenMelbz** [composer/composer#7705]
Hi,

Is there a CLI option to skip or ignore checksum verification? We're having to use a less than ideal plugin, and their satis repo is flakey.

I know you can set `checksum: false` in your satis.json but we don't control their repo and they can't change it.

Are there any options to force downloading of it regardless of the checksum verification? a bit like `--no-verify` for example?

Thanks

**Comment by alcohol** [comment]
Flakey in what sense? If connectivity is flakey, I would suggest that verifying the checksum is even more important now. It is currently not possible to disable the checksum verification in Composer.

**Comment by OwenMelbz** [comment]
Well we often get the same checksum issue, several times a day, and have to manually contact them and they do something "top secret" several hours later their end it magically fixes.

It's such a slow process thats no fault of composer/satis from what I can tell - so we were hoping there was a way we could just download the files so we can work on the project - I don't even need the package it's downloading for what I'm doing 90% of the time, but the whole install then fails.

Is there maybe instead a "continue on fail" flag? so if it cant get one package it just carries on to get the rest?

Thanks

**Comment by alcohol** [comment]
There is no such feature, no.

But you could run your own Satis instance to mirror theirs. That way, your Satis will always have the package available. And when it cannot fetch the latest one due to checksum issues, it will not affect your own development/work flow.

**Comment by Seldaek** [comment]
Alternatively you can also use packagist.com to mirror their satis repo. That should also give you better availability. Or you could suggest your plugin vendor that they use https://packagist.com/vendors to publish the package instead of rolling their own broken satis repo.

**Comment by zerustech** [comment]
@OwenMelbz I think the checksum issue was caused by a bug in `ext/phar`. I have posted a bug report at https://bugs.php.net/bug.php?id=77229

Basically, `ext/phar` generates tar archives with `pax` headers, which means even two tar archives contain the same files, they may still have different `checksums`, because they may be generated by different processes or at different time. As per the php specifications, the `ext/phar` should generate tar archives in `ustar` format.

I found this issue when trying to set up a satis server, and the workaround might be disabling checksums in `satis.json`.
